index.html

```diff
@@ -101,212 +220,1289 @@ <h3>
+              <li>
+                [=Queue a global task=] on the [=relevant global object=] of
+                [=this=] using the [=TCP socket task source=] to run the
+                following steps:
+                <ul>
+                  <li>
+                    If [=this=].{{TCPSocket/[[writable]]}} is errored,
+                    [=reject=] [=this=].{{TCPSocket/[[closedPromise]]}}
+                    with [=this=].{{TCPSocket/[[writable]]}}.`[[storedPromise]]`.
+                  <li>
+                    Otherwise, if [=this=].{{TCPSocket/[[readable]]}} is errored,
+                    [=reject=] [=this=].{{TCPSocket/[[closedPromise]]}}
+                    with [=this=].{{TCPSocket/[[readable]]}}.`[[storedPromise]]`.
+                  <li>
+                    Otherwise, [=resolve=] [=this=].{{TCPSocket/[[closedPromise]]}} with `undefined`.
+                </ul>
+            </ol>
+        </ol>
+
+      </section>
+
+      <section>
+        <h3>{{TCPSocket/[[writable]]}} attribute (internal)</h3>
+
+        <aside class="example">
+           To write individual chunks of data to the socket a
+          {{WritableStreamDefaultWriter}} can be created and released as necessary.
+          This example uses a `TextEncoder` to encode a {{DOMString}} as the
+          necessary {{Uint8Array}} for transmission.
+
+          <pre class="js">
+            const encoder = new TextEncoder();
+
+            const { writable } = await socket.opened;
+            const writer = writable.getWriter();
+
+            await writer.write(encoder.encode("PING"));  
+
+            writer.releaseLock();
+          </pre>
+
+          The {{WritableStreamDefaultWriter/write()}} method returns a {{Promise}} which
+          resolves when data has been written. While having some data available in
+          the transmit buffer is important to maintain good throughput awaiting this
+          {{Promise}} before generating too many chunks of data is a good practice
+          to avoid excessive buffering.
+        </aside>
+
+        The steps to <dfn data-lt="initialize tcp writable stream">initialize the TCP writable stream</dfn> are:
+
+        <ol>
+          <li>
+            Let |stream:WritableStream| be a [=new=] {{WritableStream}}.
+          <li>
+            Let |signal:AbortSignal| be |stream|'s [=WritableStream/signal=].
+          <li>
+            Let |writeAlgorithm| be the following steps, given |chunk|:
+            <ol>
+              <li>
+                Let |promise:Promise| be [=a new promise=].
+              <li>
+                Assert: |signal| is not [=AbortSignal/aborted=].
+              <li>
+                If |chunk| cannot be [=converted to an IDL value=] of type
+                {{BufferSource}}, reject |promise| with a {{TypeError}} and return
+                |promise|. Otherwise, save the result of the conversion in
+                |source:BufferSource|.
```

Comment:
An extra step is needed here:

> [Get a copy of the buffer source](https://webidl.spec.whatwg.org/#dfn-get-buffer-source-copy) _source_ and save the result in _bytes_.

Then we can write _bytes_ to the socket.

Reply:
Fixed.